### PR TITLE
fix: fix :empty selector.

### PR DESCRIPTION
**Pull request by Storm-M**
fix the empty bug from blank space

### CODE REVIEW

**Review by andycall — APPROVED** [review]

File: bridge/core/html/parser/html_parser.cc [hunk 1]
```diff
@@ -92,6 +93,24 @@ void HTMLParser::traverseHTML(Node* root_node, GumboNode* node) {
       } else if (child->type == GUMBO_NODE_TEXT) {
         auto* text = context->document()->createTextNode(AtomicString(ctx, child->v.text.text), ASSERT_NO_EXCEPTION());
         root_container->AppendChild(text);
+      } else if (child->type == GUMBO_NODE_WHITESPACE) {
```
Comment (XGHeaven) [inline bridge/core/html/parser/html_parser.cc:96]: Why not check GUMBO_NODE_COMMENT also?

File: bridge/core/html/parser/html_parser.cc [hunk 2]
```diff
@@ -92,6 +93,24 @@ void HTMLParser::traverseHTML(Node* root_node, GumboNode* node) {
...
+          isBlankSpace = child->v.text.text[j] == ' ';
+          if (isBlankSpace == false) {
+            break;
+          }
+        }
+
+        if (isBlankSpace) {
+          GumboTag tag = child->parent->v.element.tag;
+          if (nLen > 0) {
+            auto* comment =
```
Comment (XGHeaven) [inline bridge/core/html/parser/html_parser.cc:109]: Why convert whitespace to comment? It's not right?

File: bridge/core/html/parser/html_parser.cc [hunk 3]
```diff
@@ -92,6 +93,24 @@ void HTMLParser::traverseHTML(Node* root_node, GumboNode* node) {
...
+        int nLen = strlen(child->v.text.text);
+        for (int j = 0; j < nLen; ++j) {
+          isBlankSpace = child->v.text.text[j] == ' ';
+          if (isBlankSpace == false) {
+            break;
+          }
+        }
+
+        if (isBlankSpace) {
+          GumboTag tag = child->parent->v.element.tag;
```
Comment (XGHeaven) [inline bridge/core/html/parser/html_parser.cc:107]: unused `tag` variable.

**Review by XGHeaven — CHANGES_REQUESTED** [review]

File: integration_tests/specs/css/css-empty/empty.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  margin-left: 50px;
+  z-index: 1;
+}
+
+.demo > :last-child {
+  background: #0000ff;
+  border: 2px solid gold;
+  z-index: 4;
+  margin-left: 25px;
+}
```
[the remark this answers is not in the document]
Comment (Storm-M) [reply integration_tests/specs/css/css-empty/empty.html:35]: ok，delete it！

File: bridge/core/html/parser/html_parser.cc (hunk 1 above)
Comment (Storm-M) [reply bridge/core/html/parser/html_parser.cc:96]: it only to judge the empty: if the element only have the blank space ,  i will add textnode to  this element. or do nothing, at last  empty judgement code is to judge the element  has sub element

**Review by devjiangzhou — APPROVED** [review]

**Pull request merged**